Peoplesoft/modules/compute/compute.tf

```diff
@@ -0,0 +1,171 @@
+    storage_uri = "${var.boot_diag_SA_endpoint}"
+  }
+}
+resource "azurerm_managed_disk" "vm_data_disks" {
+    name                 = "${var.compute_hostname_prefix}-${format("%.02d",count.index + 1)}-disk-data-01"  
+    location             = "${var.location}"
+    resource_group_name  = "${var.resource_group_name}"
+    storage_account_type = "${var.storage_account_type}"
+    create_option        = "Empty"
+    disk_size_gb         = "${var.data_disk_size_gb}"
+    count                = "${var.create_data_disk}"
+
+}
+
+resource "azurerm_virtual_machine_data_disk_attachment" "vm_data_disks_attachment" {
+  managed_disk_id    = "${element(azurerm_managed_disk.vm_data_disks.*.id, count.index)}"
+  virtual_machine_id = "${element(azurerm_virtual_machine.compute.*.id, count.index)}"
+  lun                = "${count.index}"
+  caching            = "None"
+  count = "${var.create_data_disk}"
+
+
+ provisioner "remote-exec" {
+ inline = [
+ "sudo su", # TODO: Upload private key to the machine too. Figure out how. 
+ "yum install -y oracle-ebs-server-R12-preinstall.x86_64", # This will be replaced by cloud-init for Oracle Linux 7.7+ 
+ # "echo 'type=83' | sfdisk /dev/sdc", #sfdisk alternative to fdisk
+ "echo n\np\n1\n\n\np\nw | fdisk /dev/sdc", # n=> New parition, p=> primary partition, 1 => default partition number 1, "" => default first and last sector, w => write to partition table #TODO: Check if \n works
+ "mkfs -t ext4 /dec/sdc1",
+ "chown -R ${var.admin_username} /u01"
+ ]
+ } 
+}
+
+resource "azurerm_availability_set" "compute" {
+  name                         = "${var.compute_hostname_prefix}-avset"
+  location                     = "${var.location}"
+  resource_group_name          = "${var.resource_group_name}"
+  platform_fault_domain_count  = 2
+  platform_update_domain_count = 2
+  managed                      = true
+  tags                         = "${var.tags}"
+}
+
+resource "azurerm_network_interface" "compute" {
+  count                         = "${var.compute_instance_count}"
+  name                          = "${var.compute_hostname_prefix}-${format("%.02d",count.index + 1)}-nic"  
+  location                      = "${var.location}"
+  resource_group_name           = "${var.resource_group_name}"
+  enable_accelerated_networking = "${var.enable_accelerated_networking}"
+
+  ip_configuration {
+    name                          = "ipconfig${count.index}"
+    subnet_id                     = "${var.vnet_subnet_id}"
+    private_ip_address_allocation = "Dynamic"
+  }
+
+  tags = "${var.tags}"
+}
+resource "azurerm_network_interface_backend_address_pool_association" "compute" {
+  count                   = "${var.assign_bepool * var.compute_instance_count}"  
+  network_interface_id    = "${element(azurerm_network_interface.compute.*.id, count.index)}"
+  ip_configuration_name   = "ipconfig${count.index}"
+  backend_address_pool_id = "${var.backendpool_id}"
+}
+
+resource "azurerm_public_ip" "compute" {
```

Comment:
will each machine have public IP?? I thought only the bastion host needs public ip?

Reply:
Only the bastion gets a public IP... that is switched on and off with a count.